baetyl-function-manager/function.go

```diff
@@ -52,9 +54,21 @@ func (f *Function) Call(msg *baetyl.FunctionMessage) (*baetyl.FunctionMessage, e
 
 // CallAsync calls function to handle message and return result message
 func (f *Function) CallAsync(msg *baetyl.FunctionMessage, cb func(in, out *baetyl.FunctionMessage, err error)) error {
-	item, err := f.pool.BorrowObject(context.Background())
+	var item interface{}
+	var err error
+	for i := 0; i <= f.cfg.RetryNum; i++ {
+		if i != 0 {
+			time.Sleep(f.cfg.RetryInterval)
+			f.log.Infof("function (%s) retried %d time(s)", f.cfg.Name, i)
+		}
+		item, err = f.pool.BorrowObject(context.Background())
+		if err == nil {
+			break
+		}
+	}
 	if err != nil {
-		return err
+		f.log.Errorf("function (%s) retry failed,msg lost", f.cfg.Name)
+		return nil
 	}
```

Comment:
bf := &backoff.Backoff{
			Min:    time.Millisecond * 10,
			Max:    f.cfg.Backoff.Max,
			Factor: 2,
		}
for {
ctx,cancel := context.WithTimeout(context.Background(), bf.Duration())
defer cancel()
item, err := f.pool.BorrowObject(ctx))
if err != nil && strings.Contains(err.Error(),"Timeout"){
continue
}
return item,err
}